sklearn/preprocessing/tests/test_target_encoder.py

```diff
@@ -188,29 +280,31 @@ def test_errors(y, msg):
 
 
 def test_use_regression_target():
-    """Custom target_type to avoid inferring the target type."""
+    """Check inferred and specified `target_type` on regression target."""
     X = np.array([[0, 1, 0, 1, 0, 1]]).T
-
-    # XXX: When multiclass is supported, then the following `y`
-    # is considered a multiclass problem and `TargetEncoder` will not error.
-    # type_of_target would be 'multiclass'
     y = np.array([1.0, 2.0, 3.0, 2.0, 3.0, 4.0])
-    enc = TargetEncoder()
-    msg = "Target type was inferred to be 'multiclass'"
-    with pytest.raises(ValueError, match=msg):
-        enc.fit_transform(X, y)
 
-    enc = TargetEncoder(target_type="continuous")
+    enc = TargetEncoder(cv=2)
+    enc.fit_transform(X, y)
+    assert enc.target_type_ == "multiclass"
+
+    enc = TargetEncoder(cv=2, target_type="continuous")
     enc.fit_transform(X, y)
     assert enc.target_type_ == "continuous"
 
 
-def test_feature_names_out_set_output():
+@pytest.mark.parametrize(
+    "y, feature_names",
+    [
+        ([1, 2] * 10, ["A", "B"]),
+        ([1, 2, 3] * 6 + [1, 2], ["A_1", "A_2", "A_3", "B_1", "B_2", "B_3"]),
```

Comment:
Can you add a test case for multiclass targets encoded as strings?

Reply:
Good point!